src/target/lpc546xx.c

```diff
@@ -78,6 +78,48 @@ const command_s lpc546xx_cmd_list[] = {
 	{NULL, NULL, NULL},
 };
 
+#define USE_LUT
+
+#ifdef USE_LUT
+struct lpc546xx_device {
+	uint32_t chipid;
+	const char *designator;
+	uint16_t flash_kbytes;
+	uint16_t sram123_kbytes;
+};
+
+/* Reference: "LPC546XX Product data sheet" revision 2.6, 2018
+ * Part type number encoding: LPC546xxJyyy, where yyy is flash size, KiB
+ */
+static const struct lpc546xx_device lpc546xx_devices_lut[] = {
+	{0x7f954605U, "LPC54605J256", 256, 32},
+	{0x7f954606U, "LPC54606J256", 256, 32},
+	{0x7f954607U, "LPC54607J256", 256, 32},
+	{0x7f954616U, "LPC54616J256", 256, 32},
+	{0xfff54605U, "LPC54605J512", 512, 96},
+	{0xfff54606U, "LPC54606J512", 512, 96},
+	{0xfff54607U, "LPC54607J512", 512, 96},
+	{0xfff54608U, "LPC54608J512", 512, 96},
+	{0xfff54616U, "LPC54616J512", 512, 96},
+	{0xfff54618U, "LPC54618J512", 512, 96},
+	{0xfff54628U, "LPC54628J512", 512, 96},
```

Comment:
I cannot drop the array items because the linear search on chipid breaks then. I could replace all name strings with two distinct values and hope that GCC performs const char * optimization (where it substitutes pointers to literal strings with common suffix or even identical strings altogether).

Reply:
You are *guaranteed* that optimisation which is what we mean by "same decimation"

Reply:
Wins some 114 bytes at the cost of not showing the developer which flavour of chip the BMP detects. Is there any other way to display the LPC546xx chip id from SYSCON? It's not a generic cortex-m identifier. Of course, the developer could simply stare at the chip package markings instead.

Reply:
This is no more information lost than for any other part which resolves to a series name etc - we simply don't have the Flash space for nicer output.

We can't think of another, cheap, way to have that information available and it's unlikely given a form which can be made human readable from the registers.